raster-foundry/raster-foundry
# Reorganize and remove dependencies, part 1

## Overview

This PR audits dependencies, removes a number of them, reorganizes some others, and zaps what it can to make the backsplash jar in particular and jars in general smaller.

Why is it good:

A lot of dependencies are no longer a part of projects that don't need them. This is good for CLEANLINESS. For example, anything depending on `awsSdk` was a waste, since nothing actually used it.

Also, smaller jar sizes are good because we pay for storage and because they shoud... start faster? Seems like maybe that's not true. Because they take less time to assemble! For developer productivity! I'm looking for the wins I can find, since I couldn't make backsplash small enough for lambda and have other things to do than die on this hill (though I am obviously more than happy if this is the particular hill I die on eventually).

### Checklist


### Demo

Backsplash size before: ~252mb
Backsplash size now: 182mb
Batch size before: ~299mb
Batch size now: 237mb

### Notes

The bulk of the improvements for backsplash and batch came from removing the general purpose `awsSdk` dependency, which wasn't needed and was huge. Other big improvements were getting akka out of `common` and `zap`-ing things in the backsplash assembly shade rules.

## Testing Instructions

 * reassemble batch, backsplash, the api server, and the tile server
 * console into the batch container
 * pick some jobs (probably not export, since it's broken right now in dev) and run them, they should still be fine
 * `docker-compose run --rm --entrypoint bash --service-ports api-server`
 * `docker-compose run --rm --entrypoint bash --service-ports backsplash`
 * `docker-compose run --rm --entrypoint bash --service-ports tile-server`
 * start up the api server: `java -jar app-backend/api/target/scala-2.11/api-assembly.jar`
 * bring up the nginx-api container
 * wander around the app in the frontend -- it should be fine
 * start up the tile server: `java -jar app-backend/tile/target/scala-2.11/tile-assembly.jar`
 * bringup the nginx-tiler container
 * make sure tile requests still work normally and in the lab
 * copy one of each of those requests -- you'll need it in a minute
 * stop the tile server
 * start up backsplash: `java -jar app-backend/backsplash/target/sscala-2.11/backsplash-assembly.jar`
 * hit backsplash with the two requests you just copied
 * everything should have worked

Closes azavea/raster-foundry-platform#533
